packages/eslint-plugin/src/rules/strict-enums.ts

```diff
@@ -0,0 +1,685 @@
+import * as util from '../util';
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+import { TSESTree } from '@typescript-eslint/utils';
+
+/** These operators are always considered to be safe. */
+const ALLOWED_ENUM_OPERATORS = new Set(['in', '|', '&', '|=', '&=']);
+
+const ALLOWED_TYPES_FOR_ANY_ENUM_ARGUMENT =
+  ts.TypeFlags.Any |
+  ts.TypeFlags.Unknown |
+  ts.TypeFlags.Number |
+  ts.TypeFlags.String;
+
+export type Options = [];
+export type MessageIds =
+  | 'incorrectIncrement'
+  | 'mismatchedAssignment'
+  | 'mismatchedComparison'
+  | 'mismatchedFunctionArgument';
+
+export default util.createRule<Options, MessageIds>({
+  name: 'strict-enums',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallows the usage of unsafe enum patterns',
+      recommended: 'error',
```

Comment:
Rules cannot be made recommended without a breaking change


```suggestion
      recommended: false,
```

Reply:
should be fixed now